src/plots/cartesian/axes.js

```diff
@@ -362,10 +363,23 @@ axes.doAutoRange = function(ax) {
         // doAutoRange will get called on fullLayout,
         // but we want to report its results back to layout
 
-        var axIn = ax._input;
+        axIn = ax._input;
         axIn.range = ax.range.slice();
         axIn.autorange = ax.autorange;
     }
+
+    if(ax._anchorAxis && ax._anchorAxis.rangeslider) {
+        var axeRangeOpts = ax._anchorAxis.rangeslider[ax._name];
+        if(axeRangeOpts) {
+            if(axeRangeOpts.rangemode === 'auto') {
+                axeRangeOpts.range = hasDeps ? axes.getAutoRange(ax) : ax.range.slice();
+            } else {
+                axeRangeOpts.range = ax.range.slice();
+            }
+        }
+        axIn = ax._anchorAxis._input;
+        axIn.rangeslider[ax._name] = Object.assign({}, axeRangeOpts);
```

Comment:
Please use `Lib.extendFlat` instead of `Object.assign` which isn't IE-compatible.

Reply:
Indeed, completly forgot this since I'm mainly using Typescript (or ES2017 with Babel/polyfills).